crates/analyzer-host/src/lsp_impl/active_initialized.rs

```diff
@@ -229,18 +229,58 @@ async fn on_set_trace(_: LspServerState, params: SetTraceParams, state: Arc<Asyn
 	Ok(())
 }
 
+async fn created_file(url: &Url, state: &Arc<AsyncRwLock<State>>) {
+	// incase it's a folder, get a list of files
+ 	let files = state.write().await.file_system.enumerate_folder(url.clone(), "*.p4".into()).await;
+
+	// get files content from system
+	let mut content = Vec::new();
+	for url in &files {
+		content.push(fs::read_to_string(url.uri.path()).expect("Unable to read file"));
+	}
+	
+	// claim lock
+	let lock = state.write().await;
+	let mut analyzer = lock.analyzer.unwrap();
+
+	// update files
+	for i in 0..files.len()  {
+		let file_id = analyzer.file_id(files[i].uri.as_str());
+		analyzer.update(file_id, content[i].clone());
+		info!("{} file updated from file system", files[i].uri.path());
+	}
+}
+
+async fn deleted_file(url: &Url, state: &Arc<AsyncRwLock<State>>) {
+	// incase it's a folder, get a list of files
+	let files = state.write().await.file_system.enumerate_folder(url.clone(), "*.p4".into()).await;
```

Comment:
There's a `const` named `RELATIVE_P4_SOURCEFILES_GLOBPATTERN` that defines the glob pattern for `'.p4'` files. However, see my previous comment.

https://github.com/p4lang/p4analyzer/blob/8056ffcfc3cbee609d84653549c37d439d000479/crates/analyzer-host/src/lsp.rs#L32